modules/programs/sapling.nix

```diff
@@ -0,0 +1,81 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.sapling;
+
+  iniFormat = pkgs.formats.ini { };
+
+in
+{
+  meta.maintainers = [ maintainers.pbar ];
+
+  options = {
+    programs.sapling = {
+      enable = mkEnableOption "Sapling";
+
+      package = mkPackageOption pkgs "sapling" { };
+
+      userName = mkOption {
+        type = types.str;
+        description = "Default user name to use.";
+      };
+
+      userEmail = mkOption {
+        type = types.str;
+        description = "Default user email to use.";
+      };
+
+      aliases = mkOption {
+        type = types.attrsOf types.anything;
+        default = { };
+        description = "Sapling aliases to define.";
+      };
+
+      extraConfig = mkOption {
+        type = types.either (types.attrsOf types.anything) types.lines;
```

Comment:
Is there any option that cannot be represented through the attribute set alternative? Ideal would be to avoid the need for `types.lines` here.